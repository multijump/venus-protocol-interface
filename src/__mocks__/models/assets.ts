import { Asset } from 'types';

import governanceResponse from '__mocks__/api/governance.json';
import { formatToAsset } from 'clients/api/queries/getAssets';

export const assets: Asset[] = governanceResponse.data.markets
  .map(formatToAsset)
  .filter(asset => !!asset) as Asset[];
