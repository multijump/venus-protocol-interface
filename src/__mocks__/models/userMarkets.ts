import { UserMarket } from 'types';

import { userAssets } from './userAssets';

export const userMarkets: UserMarket[] = [
  {
    id: '1',
    riskLevel: 'MINIMAL',
    name: 'Venus',
    assets: userAssets,
  },
  {
    id: '2',
    riskLevel: 'VERY_HIGH',
    name: 'Gaming',
    assets: userAssets,
  },
  {
    id: '3',
    riskLevel: 'MEDIUM',
    name: 'MetaVerse',
    assets: userAssets,
  },
];
