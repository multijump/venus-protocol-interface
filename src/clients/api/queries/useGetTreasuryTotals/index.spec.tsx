import React from 'react';

import { assets } from '__mocks__/models/assets';
import { vTokenBalanceTreasury } from '__mocks__/models/vTokenBalanceTreasury';
import { getAssets, useGetVTokenBalancesAll } from 'clients/api';
import renderComponent from 'testUtils/renderComponent';

import useGetTreasuryTotals, { UseGetTreasuryTotalsOutput } from '.';

jest.mock('clients/api');

describe('api/queries/useGetTreasuryTotals', () => {
  beforeEach(() => {
    (getAssets as jest.Mock).mockImplementation(() => ({ assets }));

    (useGetVTokenBalancesAll as jest.Mock).mockImplementation(() => ({
      data: {
        balances: vTokenBalanceTreasury,
      },
    }));
  });

  it('calculates totals correctly', async () => {
    let data: UseGetTreasuryTotalsOutput['data'];

    const CallMarketContext = () => {
      ({ data } = useGetTreasuryTotals());
      expect(data).toMatchSnapshot();
      return <div />;
    };

    renderComponent(<CallMarketContext />);
  });
});
