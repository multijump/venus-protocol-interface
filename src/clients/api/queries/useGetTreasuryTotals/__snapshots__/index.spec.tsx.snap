// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/queries/useGetTreasuryTotals calculates totals correctly 1`] = `
Object {
  "treasuryTotalAvailableLiquidityBalanceCents": "0",
  "treasuryTotalBalanceCents": "0",
  "treasuryTotalBorrowBalanceCents": "0",
  "treasuryTotalSupplyBalanceCents": "0",
}
`;

exports[`api/queries/useGetTreasuryTotals calculates totals correctly 2`] = `
Object {
  "treasuryTotalAvailableLiquidityBalanceCents": "1.01091026874669540229616270538041561527001e+28",
  "treasuryTotalBalanceCents": "331065694653.315142414173534023576484",
  "treasuryTotalBorrowBalanceCents": "865582034633700.05578966984628699999",
  "treasuryTotalSupplyBalanceCents": "1.010910267860354148705085851611465407068e+26",
}
`;
